openapi2conv/openapi2_conv.go

```diff
@@ -638,6 +641,49 @@ func FromV3Swagger(swagger *openapi3.Swagger) (*openapi2.Swagger, error) {
 	return result, nil
 }
 
+func consumesToArray(consumes map[string]bool) []string {
+	var consumesArr []string
+	for key, _ := range consumes {
+		consumesArr = append(consumesArr, key)
+	}
+	sort.Strings(consumesArr)
+	return consumesArr
+}
+
+func FromV3RequestBodies(name string, requestBodyRef *openapi3.RequestBodyRef, components *openapi3.Components) (*openapi2.Parameters, *openapi2.Parameters, map[string]bool, error) {
+	formParameters := openapi2.Parameters{}
+	bodyOrRefParameters := openapi2.Parameters{}
+	if ref := requestBodyRef.Ref; ref != "" {
+		bodyOrRefParameters = append(bodyOrRefParameters, &openapi2.Parameter{Ref: FromV3Ref(ref)})
+		return &bodyOrRefParameters, &formParameters, nil, nil
+	}
+
+	//Only select one formData or request body for an individual requesstBody as swagger 2 does not support multiples
+	consumes := make(map[string]bool)
+	if requestBodyRef.Value != nil {
+		for contentType, mediaType := range requestBodyRef.Value.Content {
+			//add each type to the consumes array
+			consumes[contentType] = true
+			formParams := FromV3RequestBodyFormData(mediaType)
+			if len(formParams) != 0 {
+				formParameters = formParams
+			} else {
+				paramName := name
+				if originalName, ok := requestBodyRef.Value.Extensions["x-originalParamName"]; ok {
+					json.Unmarshal(originalName.(json.RawMessage), &paramName)
+				}
+
+				r, err := FromV3RequestBody(paramName, requestBodyRef, mediaType, components)
+				if err != nil {
+					return &bodyOrRefParameters, &formParameters, consumes, err
+				}
+				bodyOrRefParameters = append(bodyOrRefParameters, r)
+			}
+		}
+	}
+	return &bodyOrRefParameters, &formParameters, consumes, nil
```

Comment:
I'm not able to see a need for returning pointers here. These types represent slices so this seems backwards to me.

Feel free to edit https://github.com/getkin/kin-openapi/blob/fc3f90d8183b71442d5730b8e6cff877c9b93e45/openapi2/openapi2.go#L168 if it helps (please try), into:
```go
type Parameters = []*Parameter
```
NOTE: I'm not sure why it's `[]*Parameter` instead of `[]Parameter` but there are other types like this and it's surely a matter for another PR anyway.